test/runtests.jl

```diff
@@ -511,6 +518,17 @@ end
         @test_throws TypeError ParamV1(; i=1.0)
         @test_throws ArgumentError ParamV1{Int}(; i=1.1)
     end
+
+    @testset "equality and hashing" begin
+        c = ChildV1(; x=[1, 2], y="hello", z=missing)
+        @test isequal(c, c)
+        @test ismissing(c == c)
+        @test hash(c) isa UInt  # NOTE: can't rely on particular values
+        @test UnionMissingV1(; a=missing, b=1) != UnionMissingV1(; a=missing, b=2)
+        @test !isequal(UnionMissingV1(; a=missing, b=1), UnionMissingV1(; a=missing, b=2))
+        @test ParamV1(; i=one(Int32)) == ParamV1(; i=one(Int64))
+        @test isequal(ParamV1(; i=one(Int32)), ParamV1(; i=one(Int64)))
+    end
```

Comment:
Should add some tests comparing `ChildV1` to a `ParentV1` and a `ParentV1` to a `ChildV1`. In particuar I am thinking about `_compare_fields(eq, ::AbstractRecord, ::AbstractRecord)`

Reply:
Is what I just added what you had in mind?